packages/wasm-crypto/test/all/vrf.js

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019-2021 @polkadot/wasm-crypto authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+/* eslint-disable camelcase */
+
+const crypto = require('crypto');
+const { assert, hexToU8a, stringToU8a, u8aToHex } = require('@polkadot/util');
+
+function extractKeys (pair) {
+  return [pair, pair.slice(64), pair.slice(0, 64)];
+}
+
+function randomPair (wasm) {
+  return extractKeys(wasm.sr25519KeypairFromSeed(crypto.randomBytes(32)));
+}
+
+function vrf_signAndVerify (wasm) {
+  const [, pk, sk] = randomPair(wasm);
+  const outAndProof = wasm.vrfSign(sk, stringToU8a('my VRF context'), stringToU8a('this is a message'));
+  const isValid = wasm.vrfVerify(pk, stringToU8a('my VRF context'), stringToU8a('this is a message'), outAndProof);
+
+  console.log('\tVRF', u8aToHex(outAndProof));
+  console.log('\tRES', isValid);
+
+  assert(isValid, 'ERROR: Unable to verify VRF output & proof');
+}
```

Comment:
Sorry about the crappy & hacky test setup here - since it has to run in different contexts, straight-with-wasm, straight-with-asm and jest as well, it is really not the prettiest setup ever.